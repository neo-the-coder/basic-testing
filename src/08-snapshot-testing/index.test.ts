// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const elements = ['eat', 'sleep', 'code'];
    const expectedLinkedList = {
      value: 'eat',
      next: {
        value: 'sleep',
        next: {
          value: 'code',
          next: {
            value: null,
            next: null,
          },
        },
      },
    };

    const result = generateLinkedList(elements);
    expect(result).toStrictEqual(expectedLinkedList);
    expect(result).toMatchSnapshot();
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const elements = ['eat', 'sleep', 'code'];

    const result = generateLinkedList(elements);
    expect(result).toMatchSnapshot();
  });
});
