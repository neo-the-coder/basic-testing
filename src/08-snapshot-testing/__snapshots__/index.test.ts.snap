// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 1 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "code",
    },
    "value": "sleep",
  },
  "value": "eat",
}
`;

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "code",
    },
    "value": "sleep",
  },
  "value": "eat",
}
`;
